gprofiler/metadata/application_identifiers.py

```diff
@@ -287,6 +293,35 @@ def get_app_id(self, process: Process) -> Optional[str]:
         return None
 
 
+class _JavaSparkApplicationIdentifier(_ApplicationIdentifier):
+    _JAVA_SPARK_EXECUTOR_ARG = "org.apache.spark.executor.CoarseGrainedExecutorBackend"
+    _SPARK_PROPS_FILE = os.path.join("__spark_conf__", "__spark_conf__.properties")
+    _APP_NAME_NOT_FOUND = "SPARK_APP_NAME_NOT_FOUND"
```

Comment:
appids are user friendly. I expect this to look like `java: spark: app name not found`

Reply:
Waiting until you and @liorgorb will resolve the `spark: *` stuff, for now: `spark app name not found`.